File: src/main/java/com/vehicletrackingsystem/service/PositionService.java
package com.vehicletrackingsystem.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.vehicletrackingsystem.domain.Position;
import com.vehicletrackingsystem.dto.PositionDTO;
import com.vehicletrackingsystem.dto.PositionSearchDTO;
import com.vehicletrackingsystem.dto.PositionPageDTO;
import com.vehicletrackingsystem.dto.PositionConvertCriteriaDTO;
import com.vehicletrackingsystem.service.GenericService;
import com.vehicletrackingsystem.dto.common.RequestDTO;
import com.vehicletrackingsystem.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PositionService extends GenericService<Position, Integer> {

	List<Position> findAll();

	ResultDTO addPosition(PositionDTO positionDTO, RequestDTO requestDTO);

	ResultDTO updatePosition(PositionDTO positionDTO, RequestDTO requestDTO);

    Page<Position> getAllPositions(Pageable pageable);

    Page<Position> getAllPositions(Specification<Position> spec, Pageable pageable);

	ResponseEntity<PositionPageDTO> getPositions(PositionSearchDTO positionSearchDTO);
	
	List<PositionDTO> convertPositionsToPositionDTOs(List<Position> positions, PositionConvertCriteriaDTO convertCriteria);

	PositionDTO getPositionDTOById(Integer positionId);







}

File: vehicletrackingsystem_ui/src/services/PositionService.js
import http from "../http-common";

class PositionService {
  getAllPositions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/position/positions`, searchDTO);
  }

  get(positionId) {
    return this.getRequest(`/position/${positionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/position?field=${matchData}`, null);
  }

  addPosition(data) {
    return http.post("/position/addPosition", data);
  }

  update(data) {
  	return http.post("/position/updatePosition", data);
  }
  
  uploadImage(data,positionId) {
  	return http.postForm("/position/uploadImage/"+positionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PositionService();
